### PR TITLE
Flax/Jax documentation

### PR DESCRIPTION
**Pull request by mfuntowicz**
Integrating Flax & JAX into transformers documentation

### CODE REVIEW

File: src/transformers/modeling_flax_bert.py [hunk 1]
```diff
@@ -35,13 +35,19 @@
...
+    PyTorch models)
 
-    This model is also a PyTorch `torch.nn.Module <https://pytorch.org/docs/stable/nn.html#torch.nn.Module>`__
-    subclass. Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to
-    general usage and behavior.
+    This model is also a Flax Linen `flax.nn.Module
+    <https://flax.readthedocs.io/en/latest/_autosummary/flax.nn.module.html>`__ subclass. Use it as a regular Flax
+    Module and refer to the Flax documentation for all matter related to general usage and behavior.
+
+    Finally, this model supports inherent JAX features such as `Just-In-Time (JIT) compilation
```
Comment (sgugger) [inline src/transformers/modeling_flax_bert.py:46]: Would this be nicer in a list?

File: src/transformers/modeling_flax_bert.py [hunk 2]
```diff
@@ -52,50 +58,32 @@
 
 BERT_INPUTS_DOCSTRING = r"""
     Args:
-        input_ids (:obj:`torch.LongTensor` of shape :obj:`({0})`):
+        input_ids (:obj:`Numpy array` of shape :obj:`({0})`):
```
Comment (sgugger) [inline src/transformers/modeling_flax_bert.py:61]:
```suggestion
        input_ids (:obj:`numpy.ndarray` of shape :obj:`({0})`):
```

File: src/transformers/modeling_flax_bert.py [hunk 3]
```diff
@@ -52,50 +58,32 @@
...
             Indices of input sequence tokens in the vocabulary.
 
             Indices can be obtained using :class:`~transformers.BertTokenizer`. See
-            :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__` for
+            :func:`transformers.PreTrainedTokenizer.__call__` and :func:`transformers.PreTrainedTokenizer.encode` for
```
Comment (sgugger) [inline src/transformers/modeling_flax_bert.py:65]: No reason to change that line.

File: src/transformers/modeling_flax_bert.py [hunk 4]
```diff
@@ -52,50 +58,32 @@
...
             details.
 
             `What are input IDs? <../glossary.html#input-ids>`__
-        attention_mask (:obj:`torch.FloatTensor` of shape :obj:`({0})`, `optional`):
+        attention_mask (:obj:`Numpy array` of shape :obj:`({0})`, `optional`):
```
Comment (sgugger) [inline src/transformers/modeling_flax_bert.py:69]:
```suggestion
        attention_mask (:obj:`numpy.ndarray` of shape :obj:`({0})`, `optional`):
```

File: src/transformers/modeling_flax_bert.py [hunk 5]
```diff
@@ -52,50 +58,32 @@
...
             Mask to avoid performing attention on padding token indices. Mask values selected in ``[0, 1]``:
 
             - 1 for tokens that are **not masked**,
             - 0 for tokens that are **masked**.
 
             `What are attention masks? <../glossary.html#attention-mask>`__
-        token_type_ids (:obj:`torch.LongTensor` of shape :obj:`({0})`, `optional`):
+        token_type_ids (:obj:`Numpy array` of shape :obj:`({0})`, `optional`):
```
Comment (sgugger) [inline src/transformers/modeling_flax_bert.py:76]:
```suggestion
        token_type_ids (:obj:`numpy.ndarray` of shape :obj:`({0})`, `optional`):
```

File: src/transformers/modeling_flax_bert.py [hunk 6]
```diff
@@ -52,50 +58,32 @@
...
             Segment token indices to indicate first and second portions of the inputs. Indices are selected in ``[0,
             1]``:
 
             - 0 corresponds to a `sentence A` token,
             - 1 corresponds to a `sentence B` token.
 
-            `What are token type IDs? <../glossary.html#token-type-ids>`_
-        position_ids (:obj:`torch.LongTensor` of shape :obj:`({0})`, `optional`):
+            `What are token type IDs? <../glossary.html#token-type-ids>`__
+        position_ids (:obj:`Numpy array` of shape :obj:`({0})`, `optional`):
```
Comment (sgugger) [inline src/transformers/modeling_flax_bert.py:84]:
```suggestion
        position_ids (:obj:`numpy.ndarray` of shape :obj:`({0})`, `optional`):
```

File: src/transformers/modeling_flax_roberta.py [hunk 7]
```diff
@@ -51,50 +57,32 @@
 
 ROBERTA_INPUTS_DOCSTRING = r"""
     Args:
-        input_ids (:obj:`torch.LongTensor` of shape :obj:`({0})`):
+        input_ids (:obj:`Numpy array` of shape :obj:`({0})`):
```
Comment (sgugger) [inline src/transformers/modeling_flax_roberta.py:60]: Same in this file: Numpy array -> numpy.ndarray in all docstrings.

**Review by sgugger — APPROVED** [review]
Thanks for fixing the docs!

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**